toolchain/check/import_cpp.cpp

```diff
@@ -308,16 +296,8 @@ class CarbonClangDiagnosticConsumer : public clang::DiagnosticConsumer {
 static auto GenerateAst(Context& context,
                         llvm::ArrayRef<Parse::Tree::PackagingNames> imports,
                         llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem> fs,
-                        const std::string& clang_path,
-                        const std::string& target)
+                        std::shared_ptr<clang::CompilerInvocation> invocation)
     -> std::pair<std::unique_ptr<clang::ASTUnit>, bool> {
-  // Build the options to use to invoke the Clang frontend.
-  std::shared_ptr<clang::CompilerInvocation> invocation =
-      BuildCompilerInvocation(context, fs, clang_path, target);
```

Comment:
I assume this is where the `std::shared_ptr` came from originally.... do we really need a shared ptr everywhere, or could we eliminate it (or localize it to just this routine)? Hmm, maybe that's hard/impossible to do with `std::shared_ptr`....

Reply:
`clang::ASTUnit::LoadFromCompilerInvocation` takes a `shared_ptr`, so we don't have much choice at the moment. After #5543 removes the use of `ASTUnit`, we might be able to get rid of the `shared_ptr` (I've not checked but I think it's just `ASTUnit` itself that's imposing this on us, not clang more generally).